### ISSUE TITLE
Full support for tpcds queries

### ISSUE THREAD

**Issue by djouallah** [lakehq/sail#197]
so far it manage to run only 14 % of the queries

https://colab.research.google.com/drive/1NGFqooYmbKs2ffQkTzMu_AWTHCHbE419#scrollTo=di7wsjAmOABJ

**Comment by shehabgamin** [comment]
@djouallah Thank you for taking the time to evaluate Sail's support for TPC-DS queries! I'm curious if the result would differ when running the code from the main branch instead of the PyPi Python package, as we've recently merged significant improvements for handling aggregation queries.

Reaching 100% TPC-DS coverage is definitely a top priority for us, and if you're interested, we'd be thrilled to have your contributions!

**Comment by djouallah** [comment]
nice improvement, now it is 47%

<img width="338" alt="image" src="https://github.com/user-attachments/assets/bbf88120-ee25-4100-9a27-89fe5c5b5139">

**Comment by shehabgamin** [comment]
> nice improvement, now it is 47%
> 
> 
> 
> <img width="338" alt="image" src="https://github.com/user-attachments/assets/bbf88120-ee25-4100-9a27-89fe5c5b5139">
> 
> 

Thank you for tracking this! I'll take a look into the "cannot resolve attribute" errors, it's possible that there is a common theme there that will greatly increase coverage once resolved.

Side note, are you doing an installation from source for this benchmark? If not, the total duration should be much lower as it will have Sail optimized for your hardware architecture.
```
env RUSTFLAGS="-C target-cpu=native" pip install pysail -v --no-binary pysail
```

**Comment by shehabgamin** [comment]
Fixed 1 issue with qualifiers here:
https://github.com/lakehq/sail/pull/254

Will setup derived tpcds locally on Monday (PST) and measure the impact!

**Comment by linhr** [comment]
We have released v0.1.6 that fixes a few issues in TPC-DS queries. We also started #256 so that TPC-H and TPC-DS queries can be run in unit tests. We'll continue tackling issues and improve the TPC-DS coverage.

**Comment by shehabgamin** [comment]
Support for 74/99 in https://github.com/lakehq/sail/pull/261

**Comment by shehabgamin** [comment]
We support 79/99 now!

**Comment by linhr** [comment]
Support for 98/99 is now possible after #346!

**Comment by linhr** [comment]
There are a few queries that have incorrect results. I've created a separate issue for it: #347.

Update: #347 has been fixed.

**Comment by linhr** [comment]
We now have full parity with TPC-DS queries!

**Comment by djouallah** [comment]
are you sure

<img width="290" alt="Image" src="https://github.com/user-attachments/assets/8f3ccee0-76e9-49c4-bf1a-81e7b9ffecf9" />

**Comment by shehabgamin** [comment]
@djouallah Thank you for surfacing this!

It seems that the set of queries you are using in the notebook is different from the set we are testing against (https://github.com/lakehq/sail/tree/main/python/pysail/data/tpcds/queries).

We will take a look at the queries that are failing in your set and ensure we provide coverage for them.

**Comment by linhr** [comment]
Update:
1. The errors in `q21`, `q34`, and `q73` are related to decimal arithmetic and type coercion. I got errors such as `Arrow error: Invalid argument error: Invalid comparison operation: Decimal128(14, 4) > Decimal128(19, 7)` when running those queries locally. (It's unclear to me why the "operation not found" error masks the actual error in this case.) This issue also exists in DataFusion so we filed an issue upstream: <https://github.com/apache/datafusion/issues/15174>.
2. The errors in `q70` and `q86` are related to aggregation expressions in `ORDER BY` clauses for projections with a window.

The issues were not detected in our tests before since our variants of the TPC-DS queries are slightly different:
1. For `q21`, `q34`, and `q73`, the queries do not have the `* 1.000` part in expressions.
2. For `q70` and `q86`, the aggregation expressions are referred to as an alias rather than using the original aggregation.

We'll continue investigation and make sure both issues are fixed on our side.

**Comment by djouallah** [comment]
currently only q70 and q86  are not working, good progress
